pkg/api-server/gui_handler.go

```diff
@@ -29,29 +32,39 @@ var disabledPage = `
 </html>
 `
 
-func guiHandler(path string, enabledGui bool, apiUrl string, basePath string) http.Handler {
+type GuiConfig struct {
+	ApiUrl      string `json:"apiUrl"`
+	BaseGuiPath string `json:"baseGuiPath"`
+	Version     string `json:"version"`
+}
+
+func NewGuiHandler(guiPath string, enabledGui bool, guiConfig GuiConfig) (http.Handler, error) {
 	if enabledGui {
-		return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
-			if request.URL.Path == path+"config.json" {
-				writer.WriteHeader(http.StatusOK)
-				writer.Header().Add("Content-Type", "application/json")
-				_ = json.NewEncoder(writer).Encode(struct {
-					ApiUrl      string `json:"apiUrl"`
-					BaseGuiPath string `json:"baseGuiPath"`
-				}{
-					ApiUrl:      apiUrl,
-					BaseGuiPath: basePath,
-				})
+		guiFs := resources.GuiFS()
+		tmpl := template.Must(template.New("index.html").Funcs(sprig.HtmlFuncMap()).ParseFS(guiFs, "index.html"))
+		buf := bytes.Buffer{}
+		err := tmpl.Execute(&buf, guiConfig)
+		if err != nil {
+			return nil, err
+		}
+		return http.StripPrefix(guiPath, http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
```

Comment:
don't you also need an extra if for `index.html`. If someone executes `:5681/gui/index.html`, won't they get `index.html` before templating?

Reply:
No because the file server returns a 302 redirect to "/" when asked for index.html

Reply:
See: https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/net/http/fs.go;l=593-596

Reply:
interesting, thanks